superset/common/query_context_processor.py

```diff
@@ -461,7 +464,17 @@ def get_data(self, df: pd.DataFrame) -> Union[str, List[Dict[str, Any]]]:
                     df, index=include_index, **config["CSV_EXPORT"]
                 )
             elif self._query_context.result_format == ChartDataResultFormat.XLSX:
-                result = excel.df_to_excel(df, **config["EXCEL_EXPORT"])
+                ndf = df.copy()
+                columns = list(df)
+                i = 0
+                for col in columns:
+                    coltype = None
+                    if i < len(coltypes):
+                        coltype = coltypes[i]
+                        if coltype == GenericDataType.NUMERIC:
+                            ndf[col] = pd.to_numeric(df[col], errors="ignore")
+                    i += 1
```

Comment:
Do we really need to be prepared for a case where `len(coltypes != len(columns)`? AFAIK this should never happen, and if it does, I wouldn't be comfortable setting column types based on the respective index in `coltypes`..

Reply:
your call.
Agrred it should not happen "from what I've read" (which is not much :) ),
not knowing the big picture I tend to oversecure improbable situations.